packages/devtools_app/lib/src/memory/memory_treemap.dart

```diff
@@ -0,0 +1,334 @@
+// Copyright 2020 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:ui';
+
+import 'package:flutter/gestures.dart';
+import 'package:flutter/material.dart' hide TextStyle;
+import 'package:flutter/rendering.dart' hide TextStyle;
+import 'package:flutter/widgets.dart' hide TextStyle;
+import 'package:provider/provider.dart';
+
+import '../auto_dispose_mixin.dart';
+import '../treemap/treemap.dart';
+import '../trees.dart';
+
+import 'memory_controller.dart';
+import 'memory_graph_model.dart';
+import 'memory_utils.dart';
+
+class TreemapSizeAnalyzer extends SingleChildRenderObjectWidget {
+  const TreemapSizeAnalyzer({
+    Key key,
+    Widget child,
+  }) : super(key: key, child: child);
+
+  @override
+  RenderFlameChart createRenderObject(BuildContext context) {
+    return RenderFlameChart();
+  }
+}
+
+class RenderFlameChart extends RenderProxyBox {
+  @override
+  void paint(PaintingContext context, Offset offset) {
+    if (child != null) {
+      context.paintChild(child, offset);
+    }
+  }
+}
+
+class MemoryTreemap extends StatefulWidget {
+  const MemoryTreemap(
+    this.controller,
+  );
+
+  final MemoryController controller;
+
+  @override
+  MemoryTreemapState createState() => MemoryTreemapState(controller);
+}
+
+class MemoryTreemapState extends State<MemoryTreemap> with AutoDisposeMixin {
+  MemoryTreemapState(this.controller);
+
+  InstructionsSize sizes;
+
+  Map<String, Function> callbacks = {};
+
+  MemoryController controller;
+
+  Widget snapshotDisplay;
+
+  @override
+  void initState() {
+    super.initState();
+  }
+
+  @override
+  void didChangeDependencies() {
+    super.didChangeDependencies();
+
+    // TODO(terry): Unable to short-circuit need to investigate why?
+    controller = Provider.of<MemoryController>(context);
+
+    cancel();
+
+    addAutoDisposeListener(controller.selectedSnapshotNotifier, () {
+      setState(() {
+        controller.computeAllLibraries(true, true);
+
+        sizes = InstructionsSize.fromSnapshop(controller);
+      });
+    });
+
+    addAutoDisposeListener(controller.filterNotifier, () {
+      setState(() {
+        controller.computeAllLibraries(true, true);
+      });
+    });
+    // TODO(peterdjlee): Need to check if applicable to treemap.
```

Comment:
@terrylucas any thoughts on this?

Reply:
Search should be available from everywhere even treemap. Especially if the name can't be displayed at a zoomed out level e.g., navigation zero.

Search when treemap is visible should at least select the class name or library found?  But, I suspect it would highlight, select (click) and show current state in navigation bar.

Does Treemap expose a matchNames, findNode and selectNode?  If so, then I can add back this functionality post your commit.

Reply:
Let's hook the callbacks up as a follow up to this CL (which we will land after releasing next week so this won't break any users)